### PR TITLE
Bump to Go 1.21

### PR DESCRIPTION
**Pull request by jpillora**
- Bump deps
- Bump go
- Key file/gen (thanks @cmenginnz)

### CODE REVIEW

**Comment by jpillora** [comment]
Can you plz review @cmenginnz ?

File: client/client.go [hunk 1]
```diff
@@ -105,7 +105,7 @@ func NewClient(c *Config) (*Client, error) {
 		tlsConfig: nil,
 	}
 	//set default log level
-	client.Logger.Info = c.Verbose
+	client.Logger.Info = true
```
Comment (jpillora) [inline client/client.go:108]: im reverting this, users who want to silence chisel can >/dev/null

**Comment by jpillora** [comment]
I’m thinking of switching keygenfile to use plain base 64, because PEM encoding is multi line and adds an extra step when passing it inline. I’m thinking `ck-<n base64 chars>` which would also make it easier to identify its inline - rather than a file path

**Review by cmenginnz — APPROVED** [review]
LGTM

**Comment by cmenginnz** [comment]
> I’m thinking of switching keygenfile to use plain base 64, because PEM encoding is multi line and adds an extra step when passing it inline. I’m thinking `ck-<n base64 chars>` which would also make it easier to identify its inline - rather than a file path

The `ssh.ParsePrivateKey(key)` function actually supports RSA (PKCS#1), PKCS#8, DSA (OpenSSL), and ECDSA private keys. If we use a private file format here, we will lose the support of these algorithms and the existing tools such as `openssl`. 
Please let me know your decision and I'm willing to continue patching for it.

**Comment by jpillora** [comment]
I think we'd want the key parser to support both PEM and chisel-key format (`ck-<163-base64-chars>` drop the 2 equals)

**Comment by cmenginnz** [comment]
@jpillora I made a PR(https://github.com/jpillora/chisel/pull/441) to go121 branch that implements:
1. `--genkey` generates a key file in CK format
2. `--keyfile` accepts:
2.1  a file in CK format
2.2 a file in PEM formet
2.3 an inline key in CK format

**Comment by jpillora** [comment]
It took a while, but its in! Thanks for your help @cmenginnz 😄

**Comment by jpillora** [comment]
`v1.9.0` releasing now https://github.com/jpillora/chisel/actions/runs/5908510975

**Pull request merged**